File: b/models/Course.js
module.exports = (sequelize, DataTypes) => {
  const Course = sequelize.define('Course', {
    name: { type: DataTypes.STRING, allowNull: false },
    instructor: { type: DataTypes.STRING },
    category: { type: DataTypes.STRING },
    complexity: { type: DataTypes.ENUM('Core', 'Standard', 'Advanced', 'Experimental'), defaultValue: 'Core' },
    image: { type: DataTypes.STRING },
    rating: { type: DataTypes.FLOAT, defaultValue: 4.5 },
    studentsCount: { type: DataTypes.INTEGER, defaultValue: 0 },
    description: { type: DataTypes.TEXT },
  });
  return Course;
};

File: nexus-spring-backend/src/main/java/com/nexus/backend/model/Course.java
package com.nexus.backend.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "Courses")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String instructor;
    private String category;
    private String complexity; // Using String for simplicity, can be Enum
    private String image;
    private Float rating;
    private Integer studentsCount;

    @Column(columnDefinition = "TEXT")
    private String description;
}
